clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -825,6 +825,97 @@ protected void produceToLocalKafka(
+        if (rewindStartTimestamp > 0) {
+          PubSubTopicPartition newSourceTopicPartition =
+              new PubSubTopicPartitionImpl(newSourceTopic, sourceTopicPartition.getPartitionNumber());
+          try {
+            upstreamStartOffset =
+                getTopicPartitionOffsetByKafkaURL(sourceKafkaURL, newSourceTopicPartition, rewindStartTimestamp);
+            LOGGER.info(
+                "Get upstreamStartOffset: {} for source URL: {}, topic: {}, rewind timestamp: {}",
+                upstreamStartOffset,
+                sourceKafkaURL,
+                newSourceTopicPartition,
+                rewindStartTimestamp);
+          } catch (Exception e) {
+            /**
+             * This is actually tricky. Potentially we could return a -1 value here, but this has the gotcha that if we
+             * have a non-symmetrical failure (like, lor1 can't talk to the ltx1 broker) this will result in a remote
+             * colo rewinding to a potentially non-deterministic offset when the remote becomes available again. So
+             * instead, we record the context of this call and commit it to a repair queue to rewind to a
+             * consistent place on the RT
+             *
+             * NOTE: It is possible that the outage is so long and the rewind so large that we fall off retention. If
+             * this happens we can detect and repair the inconsistency from the offline DCR validator.
+             */
+            unreachableBrokerList.add(sourceKafkaURL);
+            upstreamStartOffset = OffsetRecord.LOWEST_OFFSET;
+            LOGGER.error(
+                "Failed contacting broker {} when processing topic switch! for {}. Setting upstream start offset to {}",
+                sourceKafkaURL,
+                sourceTopicPartition,
+                upstreamStartOffset);
+            hostLevelIngestionStats.recordIngestionFailure();
+
+            // Add to repair queue
+            if (remoteIngestionRepairService != null) {
+              this.remoteIngestionRepairService.registerRepairTask(
+                  this,
+                  buildRepairTask(
+                      sourceKafkaURL.toString(),
+                      sourceTopicPartition,
+                      rewindStartTimestamp,
+                      partitionConsumptionState));
+            } else {
+              // If there isn't an available repair service, then we need to abort in order to make sure the error is
+              // propagated up
+              throw new VeniceException(
+                  String.format(
+                      "Failed contacting broker (%s) and no repair service available!  Aborting topic switch processing for %s. Setting upstream start offset to %d",
+                      sourceKafkaURL,
+                      sourceTopicPartition,
+                      upstreamStartOffset));
+            }
+          }
+        } else {
+          upstreamStartOffset = OffsetRecord.LOWEST_OFFSET;
```

Comment:
Is this a valid case? Stream Reprocessing might use rewind time? Since we have a log for the previous branch anyway, could you please also add a log here?

Reply:
For now I think it is unexpected, but I can put a warn logging here.

Reply:
Yup, please do.